luda/editor/client/src/fileSystem/IFileSystem.ts

```diff
@@ -2,6 +2,13 @@ import { Dirent } from "luda-editor-common";
 
 export interface IFileSystem {
   list(path: string): Promise<Array<Dirent>>;
-  read(path: string): Promise<ArrayBuffer>;
+  read(
+    path: string,
+  ): Promise<
+    | { isSuccessful: true; file: ArrayBuffer }
```

Comment:
Actually we already have the type for it. `RpcResult`. You don't have to change it (but if you want, good).
I think the name of `RpcResult` should be just `Result`

Reply:
oops
fixed with 254bb2e